### ISSUE TITLE
Test Pilot website still looks as if it's add-on is installed, after manually disabling it from "about:addons" page

### ISSUE THREAD

**Issue by SoftVision-PaulOiegas** [mozilla/testpilot#2866]
**[Affected versions]:**
- Firefox 55.0.3 and above

**[Affected Platforms]:**
- All Windows

**[Prerequisites]:**
- The "testpilot.env" string pref was manually added with "stage" value on a new clean profile in "about:config" page.
- The "extensions.webapi.testing" boolean pref was manually added with "true" value on the crated profile.
- Test Pilot stage add-on was already installed on the created profile.

**[Steps to reproduce]:**
1. Open Firefox browser with the profile from prerequisites and navigate to https://testpilot.dev.mozaws.net/experiments page.
2. Open a new tab and navigate to "about:addons" page at "Extensions" tab.
3. Choose to manually disable the Test Pilot add-on.
4. Navigate to the Test Pilot page tab and wait for the auto refresh.
5. Manually refresh the page and observe the displayed content.

**[Expected result]:**
- Step 4: You are redirected to Test Pilot Homepage after the auto refresh is done.
- Step 5: Test Pilot Homepage is reloaded.

**[Actual result]:**
- Step 4: The page is just refreshed, you still remain on /experiments page and Test Pilot add-on seems to be still installed.
- Step 5: Manual refresh doesn't change anything, Test Pilot still shows as installed.

**[Notes]:**
- The same behavior occurs even if there is or not an experiment enabled.

**Comment by chuckharmston** [comment]
The fix for this is to make sure that the store correctly receives updates to add-on installation, uninstallation, enabling, and disabling, and would be a generally positive change for our code quality and UX.

**Comment by fzzzy** [comment]
The actual bug here may be that we don't redirect from /experiments to / when the add-on is not installed.

Also, be careful to close about:addons while testing, because the "undo" ui of about:addons causes the installation state to be incorrectly reported, and there's nothing we can do about that.

**Comment by SoftVision-PaulOiegas** [comment]
All the pages correctly adapt to the add-on manual disable except the https://tstpilot.dev.mozaws.net/experiments one. When this occurs, the next scenarios are also present:
- When TxP is disabled, even if you manually navigate to the Homepage, you are auto-redirected to /experiments.
- Opening an experiment from the /experiments page wrongly allows you to enable that experiment even if TxP is disabled

Also, If you chose to remove the TxP add-on, the /experiments page is only randomly updated. And when this occurs, the TxP re-install remains in an "Installing..." loop that can be bypassed with a refresh.

**Comment by dannycoates** [comment]
@SoftVision-PaulOiegas I can't reproduce any of that locally, but I did on dev... once, now I can't anymore. I wonder if we have a caching bug?

**Comment by dannycoates** [comment]
🙄 I figured it out. The deploy to stage of the fix failed because of a transient error in the integration_tests. I subsequently merged another PR right in the middle of testing this which succeeded that had the fixed code. Please re-verify.

<img width="471" alt="screen shot 2018-03-28 at 10 35 26 am" src="https://user-images.githubusercontent.com/87619/38046199-e4041b44-3273-11e8-87f4-6f30eba223d5.png">

**Comment by SoftVision-PaulOiegas** [comment]
@dannycoates The behavior improved a bit but it's still not ok. here are the steps I performed:
1. Navigate to https://testpilot.dev.mozaws.net and install TxP 
-You are redirected to the /experiments page. 
2. Manually disable the add-on, and switch back to the /experiments page tab.
-The tab doesn't update and remains on /experiments page.
3. Click the Firefox logo on top.
-You are redirected to the TxP Homepage where the add-on appears to be not installed.
4. Manually re-enable TxP add-on.
-The homepage correctly updates to /experiments page.
5. Manually re-disable the TxP add-on.
-The /experiments page updates to TxP Homepage.

I've also noticed the next things:

6. Each tab from where Test Pilot add-on is installed, it does not updates at manual disable/enable. Nor any other /experiment pages opened from it with "Ctrl+click" or "Middle click".
7. While add-on is enabled/disabled, if you open a new tab and navigate to https://testpilot.dev.mozaws.net , that page will always correctly update depending on add-on state.
8. While add-on is enabled/disabled  if you open a new tab and navigate to https://testpilot.dev.mozaws.net/experiments, that page will never update depending on add-on state.

So this seems to be the root cause, after TxP install, you are redirected to experiments page which will never update. While if we would just refresh the https://testpilot.mozaws.net page after install, this would always update.
